content/contributing/documentation/rst_guidelines.rst

```diff
@@ -1,162 +1,1365 @@
+referencing a specific part of a RST file by using the target as an anchor. When users click the
+reference, they are taken to the part of the documentation page where the target is defined.
+
+| The definition syntax is: ``.. _target:``
+| There are two ways to reference them, both using the ``ref`` markup:
+
+#. ``:ref:`target``` creates a hyperlink to the anchor with the heading defined below as label.
+#. ``:ref:`label <target>``` creates a hyperlink to the anchor with the given label.
+
+.. important::
+   As targets are visible from the entire documentation when referenced with the ``ref`` markup,
+   the target must be named as follows:
+
+   - Separate words using **only hyphens** and avoid using slashes;
+   - Prefix the target name with the **app/section name** and the **file name**, e.g.,
+     `accounting-taxes-configuration`.
+
+.. note::
+   - Add custom anchors for all headings so they can be referenced from any documentation file or
+     within Odoo using documentation links.
+   - Notice that there is no ``_`` at the end, contrary to what is done with :ref:`external
+     hyperlinks <contributing-rst-external-hyperlinks>`.
+
+.. list-table::
+   :class: o-showcase-table
+
+   * - Please refer to the :ref:`contributing-rst-hyperlinks-guidelines` section to
+       learn more about :ref:`relative links <contributing-rst-relative-links>`.
+
+   * - .. code-block:: text
+
+          .. _contributing-rst-hyperlinks-guidelines:
+
+          Hyperlinks
+          ==========
+
+         .. _contributing-rst-relative-links:
+
+         Use relative links for internal URLs
+         ------------------------------------
+
+         Please refer to the :ref:`<contributing-rst-hyperlinks-guidelines>` section to learn more
+         about :ref:`relative links <contributing-rst-relative-links>`.
+
+.. _contributing-rst-doc-page-hyperlinks:
+
+Documentation page hyperlinks
+-----------------------------
+
+| The ``doc`` markup allows referencing a documentation page wherever it is in the file tree through
+  a relative file path.
+| There are two ways to use the markup, both using the ``doc`` markup:
+
+
+#. ``:doc:`path_to_doc_page``` creates a hyperlink to the documentation page with the title of the
+   page as label.
+#. ``:doc:`label <path_to_doc_page>``` creates a hyperlink to the documentation page with the given
+   label.
+
+.. list-table::
+   :class: o-showcase-table
+
+   * - Please refer to `the Invoicing documentation <https://example.com/doc/accounting/invoices.html>`_
+       and to `Send a pro-forma invoice <https://example.com/doc/sales/proforma.html>`_.
+
+   * - .. code-block:: text
+
+          Please refer to :doc:`this documentation <customer_invoices>` and to
+          :doc:`../sales/sales/invoicing/proforma`.
+
+.. important::
+   :ref:`Use relative links <contributing-rst-relative-links>` for documentation page hyperlinks.
+
+.. _contributing-rst-download:
+
+File download hyperlinks
```

Comment:
Are these files referenced with relative links as well? It seems like it, but it's not mentioned explicitly.

Reply:
Perfect, thanks!